.github/workflows/update-documentation.yml

```diff
@@ -0,0 +1,138 @@
+name: Update documentation
+
+on:
+  pull_request_target:
+    branches:
+      - main
+
+jobs:
+  gather-information:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: read
+    outputs:
+      chart: ${{ steps.gather-information.outputs.chart }}
+      values_changed: ${{ steps.gather-information.outputs.values_changed }}
+      failed: ${{ steps.gather-information.outputs.failed }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          token: ${{ secrets.CONTANE_BOT_TOKEN }}
+      - name: Gather pull request information
+        id: gather-information
+        shell: bash
+        run: |
+          NUM_CHANGED_CHARTS=0
+          cd charts || exit 1
+          for chart in */; do
+            # Check if values.yaml has changed
+            if git diff --name-only HEAD~1 | grep -q "$chart"; then
+              echo "$chart: No changes detected"
+              continue
+            fi
+            echo "$chart: Changes detected"
+            NUM_CHANGED_CHARTS=$((NUM_CHANGED_CHARTS + 1))
+            echo "CHANGED_CHART=$chart" >> "$GITHUB_ENV"
+          done
+          if [ "$NUM_CHANGED_CHARTS" -eq 0 ]; then
+            echo "ERROR=No charts have changed" >> "$GITHUB_ENV"
+            echo "FAILED=true" >> "$GITHUB_ENV
+          fi
+          if [ "$NUM_CHANGED_CHARTS" -gt 1 ]; then
+              echo "ERROR=Multiple charts have changed, only update one Chart per PR" >> "$GITHUB_ENV"
+              echo "FAILED=true" >> "$GITHUB_ENV
+          fi
+          if git diff --name-only HEAD~1 | grep -q "$CHANGED_CHART/values.yaml"; then
+            echo "$CHANGED_CHART: No values.yaml changes detected"
+          fi
+          echo "$CHANGED_CHART: values.yaml changes detected"
+          echo "VALUES_CHANGED=true" >> "$GITHUB_ENV"
+      # Using actions/github-scripts because using exit 1 in the script above would not provide any output
+      # Source: https://github.community/t/no-output-on-process-completed-with-exit-code-1/123821/3
+      - id: show-error
+        name: Show error
+        if: ${{ steps.gather-information.outputs.failed }}
+        uses: actions/github-script@7.0.1
+        with:
+          script: |
+            if (!${{ steps.gather-information.outputs.error }}) {
+              core.setFailed(${{ steps.gather-information.outputs.error }});
+            }
+  update-documentation:
```

Comment:
This workflow file is getting rather big and hard to read, could you add newlines between the two jobs and also between each job's steps?